### PR TITLE
Issue #273: Caching Patterns [new pattern]

### PR DESCRIPTION
**Pull request by waisuan**
Hello.

Feel free to review the code that I have committed for the new/mentioned pattern. Do let me know if you discover any issues with the commit. 

Thanks!

### CODE REVIEW

**Comment by iluwatar** [comment]
- [ ] Put `under review` badge to the pull request
- [ ] Does the example code implement the pattern correctly and follow good coding practices?
- [ ] Does the example code have enough test coverage?
- [ ] Is the example code commented well enough?
- [ ] Is the example code following JavaDoc conventions?
- [ ] Are the project [coding conventions](https://github.com/iluwatar/java-design-patterns/wiki/3.-Coding-conventions) being followed?
- [ ] Is the class diagram generated correctly?
- [ ] Is the `index.md` implemented correctly so the pattern will show correctly on the web site?

File: caching/src/main/java/com/wssia/caching/DBManager.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.UpdateOptions;
+
+/**
+ *
+ * DBManager handles the communication with the underlying data store i.e. Database. It contains the
+ * implemented methods for querying, inserting, and updating data. MongoDB was used as the database
+ * for the application.
+ *
```
Comment (iluwatar) [inline caching/src/main/java/com/wssia/caching/DBManager.java]: <p>

**Comment by iluwatar** [comment]
@waisuan I'm done reviewing. Overall the implementation looks good. When you've implemented the changes please comment on this thread and I'll check again.

**Comment by waisuan** [comment]
Hello. Thanks for the feedback. Kindly check the latest changes and let me know if there is anything else that I have missed out. Thanks.

**Comment by iluwatar** [comment]
Thank you very much for the valuable contribution!

**Pull request merged**